main.go

```diff
@@ -462,3 +487,207 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "enterprise", "git@github.com:mendersoftware/"+repo+"-enterprise"+".git")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the branch which we are going to sync
+	gitcmd = exec.Command("git", "fetch", "opensource", branchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the Enterprise branch in which to merge into, and create the PR branch
+	gitcmd = exec.Command("git", "fetch", "enterprise", branchName+":"+PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Checkout the enterprise PR branch
+	gitcmd = exec.Command("git", "checkout", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Merge the branches
+	mergeMessage := fmt.Sprintf("Merge branch '%s' into '%s'", "opensource/"+branchName, "enterprise/"+PRBranchName)
+	gitcmd = exec.Command("git", "merge", "-m", mergeMessage, "opensource/"+branchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	log.Debugf("git merge output: %s", out)
+
+	switch string(out) {
+	case "Already up to date.\n":
+		return errors.New("The branches seem to already be in sync, which they are not supposed to be, after a merge to OS!")
+	}
+
+	// Push the branch to enterprise
+	gitcmd = exec.Command("git", "push", "--set-upstream", "enterprise", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	log.Infof("Merged branch: opensource/%s/%s into enterprise/%[1]s/%s in the Enterprise repo", repo, branchName, PRBranchName)
+
+	return nil
+}
+
+type createPRArgs struct {
+	repo        string
+	prBranch    string
+	baseBranch  string
+	accessToken string
+	message     string
+	messageBody string
+	dryRun      bool
+}
+
+func createPullRequestFromEnterpriseBranch(args createPRArgs) (*github.PullRequest, error) {
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: args.accessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+
+	newPR := &github.NewPullRequest{
+		Title:               github.String(args.message),
+		Head:                github.String(args.prBranch),
+		Base:                github.String(args.baseBranch),
+		Body:                github.String(args.messageBody),
+		MaintainerCanModify: github.Bool(false),
```

Comment:
I would set this to true: it won't hurt the feelings of the bot that we take over, and we might be able to fix small things without launching a new PR.

I don't know. Do you have an opinion?

Reply:
It was for enforcing strict consistency. But sure, can turn it off if you'd like.

Reply:
Enabled.